html/includes/graphs/device/ntp_delay.inc.php

```diff
@@ -19,9 +19,19 @@
 // We only care about our device id.
 $components = $components[$device['device_id']];
 
+//Specify whether value is ms or seconds.
+if ($device['os_group'] == 'cisco') {
+    $time=' (s)';
+    $virtlabel=' Seconds';
+} else {
+    $time = ' (ms)';
+    $virtlabel = ' Milliseconds';
```

Comment:
This needs fixing at poll time. Data should be stored the same no matter the device type so you'll need to convert the awplus from ms to s before storing it. The graph changes can then be reverted.

Reply:
OK. 

>Data should be stored the same no matter the device type so you'll need to convert the awplus from ms to s before storing it.

I would argue the cisco poller config here is in the wrong though.  

I've only got an old 1841 here but even the cisco CLI reports these stats in msec:

```
r1#sh ver | incl .bin
System image file is "flash:c1841-advipservicesk9-mz.151-4.M5.bin"
r1#sh ntp associations 

  address         ref clock       st   when   poll reach  delay  offset   disp
*~10.37.105.100   10.37.105.250    4     41     64   377  1.069 -25.587  5.027
 * sys.peer, # selected, + candidate, - outlyer, x falseticker, ~ configured
r1#sh nt
r1#sh ntp ?
  associations  NTP associations
  status        NTP status

r1#sh ntp st
Clock is synchronized, stratum 5, reference is 10.37.105.100 
nominal freq is 250.0000 Hz, actual freq is 250.0122 Hz, precision is 2**24
reference time is DE823257.2C3C884F (21:08:39.172 UTC Wed Apr 18 2018)
clock offset is -25.5874 msec, root delay is 56.56 msec
root dispersion is 163.06 msec, peer dispersion is 5.02 msec
loopfilter state is 'CTRL' (Normal Controlled Loop), drift is -0.000049008 s/s
system poll interval is 64, last update was 315 sec ago.
```

Funnily enough, the cisco NTP config doesn't work for me and reports most results as zero:

```
SNMP[/usr/bin/snmpbulkwalk -v2c -c COMMUNITY -Osqn -M /opt/librenms/mibs:/opt/librenms/mibs/cisco udp:HOSTNAME:161 .1.3.6.1.4.1.9.9.168.1.2.1.1]
.*.*.*.1.2.64158 1
.*.*.*.1.3.64158 *
.*.*.*.1.4.64158 123
.*.*.*.1.5.64158 *
.*.*.*.1.6.64158 123
.*.*.*.1.7.64158 0
.*.*.*.1.8.64158 4
.*.*.*.1.9.64158 4
.*.*.*.1.10.64158 6
.*.*.*.1.11.64158 6
.*.*.*.1.12.64158 -22
.*.*.*.1.13.64158 "00 00 00 0C "
.*.*.*.1.14.64158 "00 00 00 01 "
.*.*.*.1.15.64158 "
%i�"
.*.*.*.1.16.64158 "00 00 00 0F 00 00 00 08 "
.*.*.*.1.17.64158 "62 82 23 28 15 3F E6 D2 "
.*.*.*.1.18.64158 "DE 82 35 DE 1B C6 4A 5B "
.*.*.*.1.19.64158 "DE 82 35 DE 1B 76 C4 D0 "
.*.*.*.1.20.64158 853488881
.*.*.*.1.21.64158 255
.*.*.*.1.22.64158 0
.*.*.*.1.23.64158 "ew	D"
.*.*.*.1.24.64158 "62 82 24 B0 "
.*.*.*.1.25.64158 "ew

"
.*.*.*.1.26.64158 8
.*.*.*.1.27.64158 1
.*.*.*.1.28.64158 853488881
.*.*.*.1.29.64158 2

RRD[update /opt/librenms/rrd/10.37.105.110/ntp-10.37.105.100.rrd N:4:0:0:0]


Component: 92
    Index:      64158
    Peer:       10.37.105.100:123
    Stratum:    1.3.6.1.4.1.9.9.168.1.2.1.1.9.64158  = 4
    Offset:     1.3.6.1.4.1.9.9.168.1.2.1.1.23.64158 = 0
    Delay:      1.3.6.1.4.1.9.9.168.1.2.1.1.24.64158 = 0
    Dispersion: 1.3.6.1.4.1.9.9.168.1.2.1.1.25.64158 = 0
```


Some of the other graph changes are necessary because otherwise they simply do not display correctly.

Reply:
Ah. I'm an idiot. Perhaps the Cisco output above is actually working but because everything is less than a second, it's simply reporting 0 seconds for Offset/Delay/Dispersion. 

I've had a look at the Cisco MIB and can see for whatever reason they output these things in seconds, rather than milliseconds. I imagine this is probably historic given things like slower networks 2 decades ago...

So - I can convert my Allied Telesis stats to seconds, but as above it just seems so illogical given all my stats for Offset/Delay/Dispersion are less than 100msec

Reply:
@laf  OK have made requested changes. Thanks.